### PR TITLE
Split DHT packet

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

**Comment by kpp** [comment]
Remove packet.rs.old

File: src/toxcore/dht/packet/nodes_request.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*! NodesRequest packet
+*/
+
+#![allow(warnings)]
```
Comment (kpp) [inline src/toxcore/dht/packet/nodes_request.rs]: ?

File: src/toxcore/dht/packet/nodes_response.rs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*! NodesResponse packet
+*/
+
+#![allow(warnings)]
```
Comment (kpp) [inline src/toxcore/dht/packet/nodes_response.rs]: ?

File: src/toxcore/dht/packet/macros.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+*/
+
+/*! Macros for test functions
+*/
+
+use super::*;
+
+use quickcheck::{Arbitrary, Gen};
+
+impl Arbitrary for DhtPacket {
```
Comment (kpp) [inline src/toxcore/dht/packet/macros.rs]: Remove quickcheck tests in this PR

File: src/toxcore/dht/packet.rs.old [hunk 4]
```diff
@@ -1,1629 +0,0 @@
-/*
```
Comment (kpp) [inline src/toxcore/dht/packet.rs.old]: This commit is named as `refactor(dht): remove quickcheck`. Don't remove packet.rs.old in it

File: src/toxcore/dht/packet/ping_response.rs [hunk 5]
```diff
@@ -188,28 +184,7 @@ impl PingResponsePayload {
...
-    fn ping_response_payload_check() {
-        fn with_payload(payload: PingResponsePayload) {
-            let mut buf = [0; MAX_DHT_PACKET_SIZE];
-            let (_, len) = payload.to_bytes((&mut buf, 0)).ok().unwrap();
-            let (_, decoded) = PingResponsePayload::from_bytes(&buf[..len]).unwrap();
-            assert_eq!(decoded, payload);
-        }
-        quickcheck(with_payload as fn(PingResponsePayload));
-    }
+    use super::super::*;
```
Comment (kpp) [inline src/toxcore/dht/packet/ping_response.rs]: use fullpath to module

File: src/toxcore/dht/packet/ping_request.rs [hunk 6]
```diff
@@ -180,34 +180,14 @@ impl ToBytes for PingRequestPayload {
 #[cfg(test)]
 mod tests {
     use super::*;
```
Comment (kpp) [inline src/toxcore/dht/packet/ping_request.rs]: and this

File: src/toxcore/dht/packet/ping_response.rs [hunk 7]
```diff
@@ -188,28 +184,7 @@ impl PingResponsePayload {
 #[cfg(test)]
 mod tests {
     use super::*;
```
Comment (kpp) [inline src/toxcore/dht/packet/ping_response.rs]: and this

File: src/toxcore/dht/packet/dht_request.rs [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+            gen_be_u8!(0xfe) >>
+            gen_be_u8!(0x01) >>
+            gen_be_u64!(self.id)
+        )
+    }
+}
+
+impl NatPingRequest {
+    /// Create new ping request with a randomly generated `request id`.
+    pub fn new() -> Self {
```
Comment (kurnevsky) [inline src/toxcore/dht/packet/dht_request.rs]: Do we need these functions?

File: src/toxcore/dht/packet/ping_response.rs [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+        do_gen!(buf,
+            gen_be_u8!(0x01) >>
+            gen_be_u64!(self.id)
+        )
+    }
+}
+
+impl PingResponsePayload {
+    /// An ID of the request / response.
+    pub fn id(&self) -> u64 {
```
Comment (kurnevsky) [inline src/toxcore/dht/packet/ping_response.rs]: Do we need this function?

File: src/toxcore/dht/packet/dht_request.rs (hunk 8 above)
Comment (kpp) [reply src/toxcore/dht/packet/dht_request.rs]: good question

**Comment by kurnevsky** [comment]
It's ok in tests to use `super::*`, it's even present in rust [book](https://doc.rust-lang.org/book/second-edition/ch11-01-writing-tests.html).

**Comment by kpp** [comment]
See https://github.com/alexcrichton/pkg-config-rs/issues/64 to track failed build.

**Pull request merged**